waiter/src/waiter/service_description.clj

```diff
@@ -831,23 +831,37 @@
                              (retrieve-token-update-epoch-time token (token->token-parameters token) version)))
                       (reduce utils/nil-safe-max nil))))))
 
+(defn- assoc-token-in-env
+  "Attaches the token sequence as an environment variable in the service description."
+  [service-description token-sequence]
+  (assoc-in service-description ["env" "WAITER_CONFIG_TOKEN"] (str/join "," token-sequence)))
+
+(defn- assoc-approved-run-as-requester-parameters
+  "Attaches run-as-requester parameters if the services has been approved."
+  [service-description assoc-run-as-user-approved? service-id-prefix username]
+  (let [candidate-service-description (assoc-run-as-requester-fields service-description username)
+        candidate-service-id (service-description->service-id service-id-prefix candidate-service-description)]
+    (if (assoc-run-as-user-approved? candidate-service-id)
+      (do
+        (log/debug "appending run-as-user into pre-approved service" candidate-service-id)
+        candidate-service-description)
+      service-description)))
+
 (defrecord DefaultServiceDescriptionBuilder [kv-store max-constraints-schema metric-group-mappings profile->defaults
                                              service-description-defaults]
   ServiceDescriptionBuilder
 
   (build [this user-service-description
           {:keys [assoc-run-as-user-approved? component->previous-descriptor-fns reference-type->entry
-                  service-id-prefix source-tokens username]}]
+                  service-id-prefix source-tokens token-sequence token-service-mapping username]}]
     (let [core-service-description
-          (if (get user-service-description "run-as-user")
-            user-service-description
-            (let [candidate-service-description (assoc-run-as-requester-fields user-service-description username)
-                  candidate-service-id (service-description->service-id service-id-prefix candidate-service-description)]
-              (if (assoc-run-as-user-approved? candidate-service-id)
-                (do
-                  (log/debug "appending run-as-user into pre-approved service" candidate-service-id)
-                  candidate-service-description)
-                user-service-description)))
+          (cond-> user-service-description
+            ;; each unique token permutation will create a unique service
+            ;; leverage WAITER_CONFIG_ prefixed environment variables being allowed
```

Comment:
Please rephrase this comment. I don't think that [_permutation_](https://en.wiktionary.org/wiki/permutation#English) is used accurately here.

Reply:
Updated the comment